nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -576,7 +577,7 @@ func waitUpgradeForStatus(upgradeClientSet *upgradeClientSet, component *string,
 				return nil, model.ComponentUpgradeStatus_STATUS_IN_PROGRESS, nil
 			}
 			if err != nil {
-				return status, model.ComponentUpgradeStatus_STATUS_FAILED, err
+				return nil, model.ComponentUpgradeStatus_STATUS_FAILED, err
```

Comment:
Are we sure the setting ComponentUpgradeStatus_STATUS_FAILED is the right thing to do here? Since we were not able to retrieve the status at all, we probably don't know what's the actual status